client/homebrew/editor/snippetbar/snippetbar.jsx

```diff
@@ -32,21 +37,63 @@ const Snippetbar = createClass({
 
 	getInitialState : function() {
 		return {
-			renderer : this.props.renderer
+			renderer : this.props.renderer,
+			snippets : []
 		};
 	},
 
+	componentDidMount : async function() {
+		const rendererPath = this.props.renderer == 'V3' ? 'V3' : 'Legacy';
+		const themePath    = this.props.theme ?? '5ePHB';
+		let snippets = ThemeSnippets[`${rendererPath}_${themePath}`];
+		snippets = this.compileSnippets(rendererPath, themePath, snippets);
+		this.setState({
+			snippets : snippets
+		});
+	},
+
+	componentDidUpdate : async function(prevProps) {
+		if(prevProps.renderer != this.props.renderer || prevProps.theme != this.props.theme) {
+			const rendererPath = this.props.renderer == 'V3' ? 'V3' : 'Legacy';
+			const themePath    = this.props.theme ?? '5ePHB';
+			console.log({ ThemeSnippets: ThemeSnippets });
+			let snippets = ThemeSnippets[`${rendererPath}_${themePath}`];
+			snippets = this.compileSnippets(rendererPath, themePath, snippets);
+			this.setState({
+				snippets : snippets
+			});
+		}
+	},
+
+	mergeCustomizer : function(objValue, srcValue) {
+		if(_.isArray(objValue)) {
+			const result = _.unionBy(srcValue, objValue, 'name'); // Join snippets together, with preference for the current theme over the base theme
+			return _.filter(result, 'gen'); //Only keep snippets with a 'gen' property.
+		}
+	},
+
+	compileSnippets : function(rendererPath, themePath, snippets) {
+		let compiledSnippets = snippets;
+		const baseThemePath = Themes[rendererPath][themePath].baseTheme;
+		console.log({ baseSnippets: ThemeSnippets[`${rendererPath}_${baseThemePath}`] });
+		console.log({ themeSnippets: compiledSnippets });
+
+		if(baseThemePath) {
+			compiledSnippets = _.mergeWith([], ThemeSnippets[`${rendererPath}_${baseThemePath}`], compiledSnippets, this.mergeCustomizer);
+			console.log({ compiledSnippets: compiledSnippets });
+			//this.compileSnippets(rendererPath, themePath, compiledSnippets); (for nested baseThemes)
+		}
+		return compiledSnippets;
+	},
+
 	handleSnippetClick : function(injectedText){
 		this.props.onInject(injectedText);
 	},
 
 	renderSnippetGroups : function(){
 		let snippets = [];
 
-		if(this.props.renderer === 'V3')
-			snippets = SnippetsV3.filter((snippetGroup)=>snippetGroup.view === this.props.view);
-		else
-			snippets = SnippetsLegacy.filter((snippetGroup)=>snippetGroup.view === this.props.view);
+		snippets = this.state.snippets.filter((snippetGroup)=>snippetGroup.view === this.props.view);
```

Comment:
Since there's only the one adjustment happening to the snippets object, I think you could declare snippets as a `const` instead of a `let` and have it be the same line.

Reply:
This is done.